Getting-Started-with-Akka-http-signature.md

```diff
@@ -0,0 +1,250 @@
+"""
+``` 
+This will return a String representation of the Pointed Graph in turtle format. This representation can also be saved on your local file system like so:
+
+```scala
+write(wd/"publicKey.ttl", RSAKeys.save(toTurtle))
+```
+
+As is evident, the pointer is an automatically generated Blank Node, which can be quite difficult to process. Because of this, it would be more optimal to change the pointer to not be a blank node but rather a #uri. The user can change the pointer manually by using a file editor into something simple - such as: *<#key>*
+
+### Attaching Public keys to a File/URI
+
+One can use the `cp` or the `mv` Ammonite commands to move the public key file into the test_www directory which resides within the rww-play directory. The process of attaching the file, containing the key to a URI is very similar.
+
+```scala
+cp(wd/"publicKey.ttl" , wd/"rww-play"/"test_www"/"pubKey.ttl")
+```
+
+The Rww-play library, which the Akka-http-signature library makes use of, uses symbolic links to indicate default representations of different documents. The Http content negotiation functionality is thus preserved. Http Content negotiation is a mechanism that is used for representing different representations of a resource at the same URI. The user can then specify which is best suited for them. This is quite a complex mechanism and the setup for it can vary greatly between different servers.
+
+One can make a symbolic link to their public key file using the following commands in the bash console:
+
+
+```bash
+$ ln -s pubKey.ttl pubKey
+```
+### Access Control
+
+Currently in order to manipulate the access control one can use `curl` and `PATCH` commands [As detailed in the rww-play wiki page](https://github.com/read-write-web/rww-play/wiki/Curl-Interactions).
+
+To demonstrate how access control changes user permitted actions, one can also manually edit the .acl.ttl file. To do this, one must navigate to the test_www directory (or the server that contains the Key files) and look for the correct file. After that in order to permit access to the , the user must comment out the following:
+
+```bash
+@prefix acl: <http://www.w3.org/ns/auth/acl#> .
+@prefix cert: <http://www.w3.org/ns/auth/cert#> .
+
+[] acl:accessTo <card>, <card.ttl> ;
+   acl:agent <card#me> ;
+#        acl:agentClass <http://xmlns.com/foaf/0.1/Agent> ;
+        acl:mode acl:Read .
+
+# we allow authentication on the key too
+<card#me> cert:key <key#> .
+
```

Comment:
That was wrong in the acl, so I removed it and replaced it with something that gives access to whatever agent can present the key.